install_config/topics/sssd_for_ldap_failover_configure_openshift.adoc

```diff
@@ -0,0 +1,66 @@
+////
+configure openshift to use SSSD for LDAP failover
+
+Module included in the following assemblies:
+
+* install_config/sssd_for_ldap_failover.adoc
+////
+
+[id='sssd-for-ldap-configure-openshift-{context}']
+= Configuring {product-title} to use SSSD as the basic remote authentication server
+
+Modify the default configuration of your cluster to use the new identity
+provider that you created. Complete the following steps on the first master host
+listed in the Ansible host inventory file.
+
+////
+[NOTE]
+====
+The following files describe an "all in one" configuration. 
+xref:../master_node_configuration.adoc#install-config-master-node-configuration[Master and Node
+Configuration] provides more information about other configurations. You might
+need to adjust these steps to fit your cluster.
+====
+////
+
+. Open the *_/etc/origin/master/master-config.yaml_* file.
+
+. Locate the *identityProviders* section and replace it with the following code:
++
+----
+  identityProviders:
+  - name: sssd
+    challenge: true
+    login: true
+    mappingMethod: claim
+    provider:
+      apiVersion: v1
+      kind: BasicAuthPasswordIdentityProvider
+      url: https://remote-basic.example.com/check_user.php
+      ca: /etc/origin/master/ca.crt
+      certFile: /etc/origin/master/openshift-master.crt
+      keyFile: /etc/origin/master/openshift-master.key
+----
+
+. Start {product-title} with the updated configuration:
++
+----
+# openshift start \
+    --public-master=https://openshift.example.com:8443 \
+    --master-config=/etc/origin/master/master-config.yaml \
+    --node-config=/etc/origin/node-node1.example.com/node-config.yaml
+----
+
+. Test a login by logging in by using the `oc` cli:
++
+----
+oc login https://openshift.example.com:8443
+----
++
+You can log in only with valid LDAP credentials.
+. List the identities and confirm that an email address is displayed for each
+user name. Run the following command:
++
+----
+oc get identities -o yaml
```

Comment:
@enj Seems there no email field in identity.
`# oc get identity
NAME       IDP NAME   IDP USER NAME   USER NAME   USER UID
sssd:pm5   sssd       pm5             pm5         0a6758df-6308-11e8-be63-fa163efb1a65`

Reply:
@kalexand-rh actually `identities` is probably fine as-is.  We tend to say `pods`, `groups`, etc...

Reply:
`oc get identity` is used more frequently in the collection, so I'll keep the change.

Reply:
@enj email still not visible,
`# oc get identity -o yaml
apiVersion: v1
items:
- apiVersion: user.openshift.io/v1
  extra:
    name: pm5 IPA
  kind: Identity
  metadata:
    creationTimestamp: 2018-05-31T07:05:41Z
    name: sssd:pm5
    namespace: ""
    resourceVersion: "6203"
    selfLink: /apis/user.openshift.io/v1/identities/sssd%3Apm5
    uid: 07884bdf-64a1-11e8-a911-fa163e12ab07
  providerName: sssd
  providerUserName: pm5
  user:
    name: pm5
    uid: 0787ffe5-64a1-11e8-a911-fa163e12ab07
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""`

Reply:
@enj the email still not visible,
`# oc get identity -o yaml
apiVersion: v1
items:
- apiVersion: user.openshift.io/v1
  extra:
    name: pm5 IPA
  kind: Identity
  metadata:
    creationTimestamp: 2018-05-31T07:05:41Z
    name: sssd:pm5
    namespace: ""
    resourceVersion: "6203"
    selfLink: /apis/user.openshift.io/v1/identities/sssd%3Apm5
    uid: 07884bdf-64a1-11e8-a911-fa163e12ab07
  providerName: sssd
  providerUserName: pm5
  user:
    name: pm5
    uid: 0787ffe5-64a1-11e8-a911-fa163e12ab07
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""`